pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -0,0 +1,26 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC
+
+from pytorch_lightning.strategies import Strategy
+from pytorch_lightning.utilities import rank_zero_deprecation
+
+
+class TrainingTypePlugin(Strategy, ABC):
```

Comment:
Just curious, do we need ABC here?

Reply:
This is just to make the typing and linter system happy, doesn't hurt
![image](https://user-images.githubusercontent.com/5495193/147300956-ee3c9e76-b420-4b2e-89b5-ce50fbd90f7f.png)